google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateTopicAndPublishMessages.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.examples.pubsub.snippets;
+
+import com.google.cloud.pubsub.spi.v1.Publisher;
+import com.google.cloud.pubsub.spi.v1.PublisherClient;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.TopicName;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A snippet for Google Cloud Pub/Sub showing how to create a Pub/Sub topic and asynchronously
+ * publish messages to it.
+ */
+public class CreateTopicAndPublishMessages {
+  public static void main(String... args) throws Exception {
+    TopicName topic = TopicName.create("test-project", "test-topic");
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      publisherClient.createTopic(topic);
+    }
+
+    Publisher publisher = null;
+    try {
+      publisher = Publisher.Builder.newBuilder(topic).build();
+      List<String> messages = Arrays.asList("first message", "second message");
+      List<ListenableFuture<String>> messageIds = new ArrayList<>();
+      for (String message : messages) {
+        messageIds.add(
+            publisher.publish(
+                PubsubMessage.newBuilder().setData(ByteString.copyFromUtf8(message)).build()));
+      }
+      for (String messageId : Futures.allAsList(messageIds).get()) {
```

Comment:
As discussed offline, the threshold should in principle be small, so it should never really be an issue.